src/main/java/nl/tudelft/contextproject/model/level/Room.java

```diff
@@ -0,0 +1,201 @@
+package nl.tudelft.contextproject.model.level;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.jme3.light.Light;
+
+import nl.tudelft.contextproject.model.entities.Entity;
+import nl.tudelft.contextproject.model.level.roomIO.RoomParser;
+import nl.tudelft.contextproject.model.level.util.Vec2I;
+import nl.tudelft.contextproject.util.Size;
+
+/**
+ * Class that represent a room used in level creation.
+ * 
+ * This class is a wrapper for the level reader so that it can be used with the
+ * level creation class.
+ */
+public class Room {
+
+	/**
+	 * Read format from map file.
+	 * TODO: Refactor parser classes so that size only is defined at one location.
+	 */
+	private static final Pattern PATTERN = Pattern.compile("(?<width>\\d+)x(?<height>\\d+)_.*");
+
+	/**
+	 * Room size. 
+	 */
+	public Size size;
+
+	/**
+	 * Entities in the room.
+	 */
+	public Set<Entity> entities;
+
+	/**
+	 * Lights in the room.
+	 */
+	public List<Light> lights;
+
+	/**
+	 * MazeTile map.
+	 */
+	public MazeTile[][] mazeTiles;
+
+	/**
+	 * Entrance door locations.
+	 */
+	public ArrayList<Vec2I> entranceDoorsLocations;
+
+	/**
+	 * Exit door locations.
+	 */
+	public ArrayList<Vec2I> exitDoorLocations;
+
+	/**
+	 * Original folder.
+	 */
+	public String folder;
+	
+	/**
+	 * Constructor will load room from files using RoomIO.
+	 * @param folder
+	 * 		fileName of the room
+	 */
+	public Room(String folder) {
+		this.folder = folder;
+		entities = new HashSet<Entity>();
+		lights = new ArrayList<Light>();
+		try {
+			size = getSizeFromFileName(folder);
+			mazeTiles = new MazeTile[size.getWidth()][size.getHeight()];
+			RoomParser.importFile(folder, mazeTiles, entities, lights, 0, 0);	
+			setDoors();
+		} catch (IOException e) {
+			Logger.getLogger("MazeGeneration").severe("Unable to correctly read name!");
+		}
+	}
+
+	/**
+	 * Get size of room from folder name.
```

Comment:
please add a newline after this line, so the @ lines are seperated from the method description